gui/codeeditstylecontrols.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2019 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "codeeditstylecontrols.h"
+#include <QColorDialog>
+
+SelectColorButton::SelectColorButton( QWidget* parent ) :
+    QPushButton( parent ),
+    mColor( QColor( 255, 255, 255 ))
+{
+    updateColor();
+    connect( this, SIGNAL( clicked() ), this, SLOT( changeColor() ));
+}
+
+void SelectColorButton::updateColor() {
+    QString btnColorStyle = QString(
+                                "background-color:rgb(%1,%2,%3);"
+                                "border-style:outset;"
+                                "border-width: 1px;")
+                            .arg( mColor.red() )
+                            .arg( mColor.green() )
+                            .arg( mColor.blue() );
+    setObjectName("SelectColorButton");
+    setStyleSheet( btnColorStyle );
+}
+
+void SelectColorButton::changeColor() {
+    QColorDialog *pDlg = new QColorDialog( mColor );
```

Comment:
As far as I know you don't need to use new/delete for QColorDialog in this method.

    QColorDialog dlg(mColor);